okhttp-tests/src/test/java/okhttp3/EventListenerTest.java

```diff
@@ -678,6 +773,10 @@ private void responseBodyFail(Protocol expectedProtocol) throws IOException {
         .url(server.url("/"))
         .build());
     Response response = call.execute();
+    if (expectedProtocol == Protocol.HTTP_2) {
+      // soft failure since client may not support depending on Platform
```

Comment:
Our tests should _always_ support HTTP/2. We don’t run tests in environments where that is absent.

(Maybe we should?)

Reply:
I would argue that the tests should be runnable without HTTP/2 support.  But agree it is a separate topic.  I don't see the harm in making HTTP/2 tests shortcircuit if they can't support it.  It might be cleaner to check some new method Platform.getInstance().isHttp2Supported()

Reply:
yeah, for sure. We can use `assumeTrue()` for that.